### PR TITLE
Fix some audit comments

### PR DESCRIPTION
**Pull request by pythonberg1997**

### CODE REVIEW

File: contracts/BSCValidatorSet.sol [hunk 1]
```diff
@@ -749,17 +749,20 @@ contract BSCValidatorSet is IBSCValidatorSet, System, IParamSubscriber, IApplica
...
-      if (isWorkingValidator(i)) {
-        voteAddrs[index] = validatorExtraSet[i].voteAddress;
-        index ++;
+
+    uint index;
+    uint length = validators.length;
+    bytes[] memory voteAddrs = new bytes[](length);
+    for (uint i; i<length; ++i) {
+      index = currentValidatorSetMap[validators[i]];
+      if (isWorkingValidator(index)) {
```
Comment (NashBC) [inline contracts/BSCValidatorSet.sol]: There is no need to check the validator is working validator, just assign the validator's vote address is ok.

File: contracts/BSCValidatorSet.sol [hunk 2]
```diff
@@ -749,17 +749,20 @@ contract BSCValidatorSet is IBSCValidatorSet, System, IParamSubscriber, IApplica
...
     if (validatorExtraSet.length != n) {
       return voteAddrs;
     }
-    uint index = 0;
-    for (uint i; i<n; ++i) {
-      if (isWorkingValidator(i)) {
-        voteAddrs[index] = validatorExtraSet[i].voteAddress;
-        index ++;
+
+    uint index;
```
Comment (NashBC) [inline contracts/BSCValidatorSet.sol]: Why not define index within the for loop?

File: contracts/BSCValidatorSet.sol (hunk 1 above)
Comment (pythonberg1997) [reply contracts/BSCValidatorSet.sol]: Fixed

File: contracts/BSCValidatorSet.sol (hunk 2 above)
Comment (pythonberg1997) [reply contracts/BSCValidatorSet.sol]: Fixed

**Pull request merged**